cmd/client/txgen/main.go

```diff
@@ -232,24 +251,33 @@ pushLoop:
 }
 
 // SendTxsToShard sends txs to shard, currently just to beacon shard
-func SendTxsToShard(clientNode *node.Node, txs types.Transactions) {
+func SendTxsToShard(clientNode *node.Node, txs types.Transactions, shardID uint32) {
 	msg := proto_node.ConstructTransactionListMessageAccount(txs)
-	err := clientNode.GetHost().SendMessageToGroups([]p2p.GroupID{p2p.GroupIDBeaconClient}, p2p_host.ConstructP2pMessage(byte(0), msg))
+	var err error
+	if shardID == 0 {
+		err = clientNode.GetHost().SendMessageToGroups([]p2p.GroupID{p2p.GroupIDBeaconClient}, p2p_host.ConstructP2pMessage(byte(0), msg))
+	} else {
+		clientGroup := p2p.NewClientGroupIDByShardID(p2p.ShardID(shardID))
+		err = clientNode.GetHost().SendMessageToGroups([]p2p.GroupID{clientGroup}, p2p_host.ConstructP2pMessage(byte(0), msg))
+	}
 	if err != nil {
 		utils.GetLogInstance().Debug("Error in Sending Txns", "Err", err)
 	}
 }
 
 // GenerateSimulatedTransactionsAccount generates simulated transaction for account model.
 func GenerateSimulatedTransactionsAccount(shardID uint32, node *node.Node, setting Settings) (types.Transactions, error) {
-	_ = setting // TODO: make use of settings
-	txs := make([]*types.Transaction, 100)
+	TxnsToGenerate := setting.MaxNumTxsPerBatch // TODO: make use of settings
+	txs := make([]*types.Transaction, TxnsToGenerate)
+	rounds := (TxnsToGenerate / 100)
```

Comment:
I see the reason for visible by 100. But it shouldn't be a hard requirement. It should be able to send any amount. Also why multiple rounds?

Reply:
I copied over what we had earlier. With rounds we calculate `basenonce` once and then add the `rounds` number to adjust the correct nonce.

Reply:
This code is now changed.